### PR TITLE
Update EIP-2537: Gas pricing MAP, MUL and ADD operations

### PR DESCRIPTION
**Pull request by MarekM25**
Nethermind proposal of MUL and ADD operations. The speed of this precompile measured on Intel NUC 11 Intel Core i7-1165G7 and Nethermind client (EcRecover ~80 mgas/s). Full proposal: https://github.com/Marchhill/bls-precompile-benchmarks/blob/main/proposed-changes.md

```
g1add-1: 86.61 mgas/s (83.42-90.05)
g1mul-1: 86.90 mgas/s (83.24-90.89)
g2add-1: 82.34 mgas/s (78.50-86.58)
g2mul-1: 93.40 mgas/s (88.11-99.38)
mapfp-1: 89.96 mgas/s (85.91-94.40)
```

**ATTENTION: ERC-RELATED PULL REQUESTS NOW OCCUR IN [ETHEREUM/ERCS](https://github.com/ethereum/ercs)**

--

When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: EIPS/eip-2537.md [hunk 1]
```diff
@@ -291,11 +291,11 @@ The cost of the pairing check operation is `43000*k + 65000` where `k` is a numb
 
 #### Fp-to-G1 mapping operation
 
-Fp -> G1 mapping is `5500` gas.
+Fp -> G1 mapping is `4125` gas.
 
 #### Fp2-to-G2 mapping operation
 
-Fp2 -> G2 mapping is `75000` gas
+Fp2 -> G2 mapping is `14055` gas
```
Comment (chfast) [inline EIPS/eip-2537.md:298]: Maybe let's put 15000 here as you have in your NUC benchmarks. This number of from my BLST benchmarks. Considering this was 5x higher originally and nobody complained...

Comment (MarekM25) [reply EIPS/eip-2537.md]: Updated to 17000 based on the feedback from @jwasinger

**Review by chfast — APPROVED** [review]

**Review by kevaundray — APPROVED** [review]

**Review by jwasinger — APPROVED** [review]

**Review by jwasinger — APPROVED** [review]

**Comment by garyschulte** [comment]
👍  from besu, implementing these gas costs and benchmarking on a nuc 13:
https://gist.github.com/garyschulte/c3e7a8015c8014707303e7f3a83dd117

**Comment by mratsim** [comment]
On constantine, on a Ryzen 7840U (laptop Zen4 15W~28W, 2023).

If needed, my lowest power PC that can run an Ethereum node (no Rpi4 anymore) has a i3-1220P (28W, 2021).

- g1add: 500 -> 375
- g2add: 800 -> 600
- g2mul: 45000 -> 22500
- g2map: 75000 -> 23800

```
git clone https://github.com/mratsim/constantine
cd constantine
CC=clang nimble bench_eth_evm_precompiles
```

Before
```
--------------------------------------------------------------------------------------------------------------------------------
BLS12_G1ADD                  500 gas     171.70 MGas/s      343406.593 ops/s         2912 ns/op         9592 CPU cycles (approx)
BLS12_G2ADD                  800 gas     204.08 MGas/s      255102.041 ops/s         3920 ns/op        12913 CPU cycles (approx)
BLS12_G1MUL                12000 gas     143.37 MGas/s       11947.717 ops/s        83698 ns/op       275678 CPU cycles (approx)
BLS12_G2MUL                45000 gas     335.51 MGas/s        7455.843 ops/s       134123 ns/op       441766 CPU cycles (approx)
BLS12_MAP_FP_TO_G1          5500 gas     164.65 MGas/s       29936.535 ops/s        33404 ns/op       110023 CPU cycles (approx)
BLS12_MAP_FP2_TO_G2        75000 gas     680.56 MGas/s        9074.163 ops/s       110203 ns/op       362978 CPU cycles (approx)
--------------------------------------------------------------------------------------------------------------------------------
```

After

```
--------------------------------------------------------------------------------------------------------------------------------
BLS12_G1ADD                  375 gas     128.73 MGas/s      343288.706 ops/s         2913 ns/op         9597 CPU cycles (approx)
BLS12_G2ADD                  600 gas     153.37 MGas/s      255623.722 ops/s         3912 ns/op        12887 CPU cycles (approx)
BLS12_G1MUL                12000 gas     140.73 MGas/s       11727.867 ops/s        85267 ns/op       280844 CPU cycles (approx)
BLS12_G2MUL                22500 gas     171.93 MGas/s        7641.521 ops/s       130864 ns/op       431031 CPU cycles (approx)
BLS12_MAP_FP_TO_G1          5500 gas     158.72 MGas/s       28857.530 ops/s        34653 ns/op       114139 CPU cycles (approx)
BLS12_MAP_FP2_TO_G2        23800 gas     217.35 MGas/s        9132.420 ops/s       109500 ns/op       360665 CPU cycles (approx)
--------------------------------------------------------------------------------------------------------------------------------
```

gas schedule can easily be edited here https://github.com/mratsim/constantine/blob/7cffd2f/benchmarks/bench_eth_evm_precompiles.nim#L45-L61

**Review by ralexstokes — APPROVED** [review]
lgtm

**Pull request merged**